pyomo/contrib/solver/config.py

```diff
@@ -31,21 +31,45 @@
 
 
 def TextIO_or_Logger(val):
-    ans = []
-    if not isinstance(val, Sequence):
+    """
+    Validates and converts input into a list of valid output streams.
+
+    Accepts:
+      - sys.stdout
+      - Instances of io.TextIOBase
+      - logging.Logger (wrapped as LogStream)
+      - Boolean values (`True` → sys.stdout)
+
+    Returns:
+      - A list of validated output streams.
+
+    Raises:
+      - ValueError if an invalid type is provided.
+    """
+    if isinstance(val, (sys.stdout.__class__, io.TextIOBase, logging.Logger)):
         val = [val]
+
+    elif isinstance(val, Sequence) and not isinstance(val, (str, bytes)):
+        val = list(val)
+
+    else:
+        val = [val]
+
+    ans = []
+
     for v in val:
         if v.__class__ in native_logical_types:
             if v:
                 ans.append(sys.stdout)
-        elif isinstance(v, io.TextIOBase):
+        elif isinstance(v, (sys.stdout.__class__, io.TextIOBase)):
```

Comment:
Please add a comment why we are testing against `stdout.__class__`.  It was not obvious (at least to me) that we are guarding against people (like us!) defining new file-like classes that do not derive from `TextIOBase` and then assigning them to stdout / stderr -- and we still want to accept those classes.